observation/qos/evm_response.go

```diff
@@ -0,0 +1,113 @@
+package qos
+
+import (
+	"errors"
+)
+
+var errInvalidResponseType = errors.New("endpont response observation does not match any registered response type")
+
+// EVMResponseHandler defines the interface for handling different response types
+type EVMResponseHandler interface {
```

Comment:
@adshmh I left all my other changes as comments or TODOs but leaving this one as an actionable.

Wdyt of renaming everything in this file (and the file name) to `evm_response_observers.go`.

I'm getting really confused of the responsibility of this file (`observation/qos/evm_response.go`) relative to `qos/evm/response.go`.

My assumption is that this file is should capture `responseObservationHandlers` or `observationHandlers` for data processing, where `qos` is the one responsible for actual response management.

Reply:
Thanks, this is a good callout: will do a small refactor to clarify the purpose of the structs in the package.

Reply:
Is this tracked in a TODO and/or GitHub Issue?

If not, let's make sure it is.

Reply:
Yes, this is completed: file and struct names updated in `observation/qos` package.